qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -38,31 +38,52 @@
 
 
 class DrivenControl:
-    """
-    Creates a driven control. A driven is a set of segments made up of amplitude vectors
-    and corresponding durations.
+    r"""
+    A piecewise-constant driven control for a single qubit.
 
     Parameters
     ----------
     rabi_rates : np.ndarray, optional
-        1-D array of size nx1 where n is number of segments;
-        Each entry is the rabi rate for the segment. Defaults to None.
+        The Rabi rates :math:`\{\Omega_n\}` for each segment, in units of radians per second. Every
+        element must be non-negative. Represented as a 1D array of length :math:`N`, where :math:`N`
+        is number of segments. You can omit this field if the Rabi rate is zero on all segments.
     azimuthal_angles : np.ndarray, optional
-        1-D array of size nx1 where n is the number of segments;
-        Each entry is the azimuthal angle for the segment; Defaults to None.
+        The azimuthal angles :math:`\{\phi_n\}` for each segment. Represented as a 1D array of
+        length :math:`N`, where :math:`N` is number of segments. You can omit this field if the
+        azimuthal angle is zero on all segments.
     detunings : np.ndarray, optional
-        1-D array of size nx1 where n is the number of segments;
-        Each entry is the detuning angle for the segment; Defaults to None.
+        The detunings :math:`\{\Delta_n\}` for each segment, in units of radians per second.
+        Represented as a 1D array of length :math:`N`, where :math:`N` is number of segments. You
+        can omit this field if the detuning is zero on all segments.
     durations : np.ndarray, optional
-        1-D array of size nx1 where n is the number of segments;
-        Each entry is the duration of the segment (in seconds); Defaults to None.
+        The durations :math:`\{\delta t_n\}` for each segment, in units of seconds. Every element
+        must be positive. Represented as a 1D array of length :math:`N`, where :math:`N` is number
+        of segments. Defaults to an array of ones if omitted.
     name : string, optional
-        An optional string to name the driven control. Defaults to None.
+        An optional string to name the control. Defaults to ``None``.
 
     Raises
```

Comment:
shall we remove this? as we don't have it for BO and this error seems obvious.

Reply:
IIRC pylint forced me to add it. I personally think it's a stupid check (since it doesn't work transitively), but while it's enabled we should abide by it.